wfe/wfe.go

```diff
@@ -1679,15 +1660,6 @@ func (wfe *WebFrontEndImpl) Challenge(
 	logEvent *requestEvent,
 	response http.ResponseWriter,
 	request *http.Request) {
-
-	// Unauthenticated GETs to challenges are only allowed when not running in
-	// strict mode.
```

Comment:
I think the diff to this function missed some complexity that could be reduced further down at L1700 (Github won't let me leave a comment on the line):
```
	// If there was an account authenticating this GET request then make sure it
	// owns the challenge.
	if account != nil && chal.Authz.Order.AccountID != account.ID {
```
There shouldn't be any case where `account == nil` with mandatory POST-as-GET right?